src/main/java/pl/tomaszdziurko/jvm_bloggers/view/admin/moderation/BlogPostItemPopulator.java

```diff
@@ -0,0 +1,80 @@
+package pl.tomaszdziurko.jvm_bloggers.view.admin.moderation;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.wicket.AttributeModifier;
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.form.Form;
+import org.apache.wicket.markup.html.link.ExternalLink;
+import org.apache.wicket.markup.html.panel.EmptyPanel;
+import org.apache.wicket.markup.repeater.Item;
+import org.apache.wicket.model.AbstractReadOnlyModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPost;
+import pl.tomaszdziurko.jvm_bloggers.utils.NowProvider;
+import pl.tomaszdziurko.jvm_bloggers.view.panels.CustomFeedbackPanel;
+
+import java.time.format.DateTimeFormatter;
+
+import static org.apache.commons.lang3.StringUtils.abbreviate;
+import static pl.tomaszdziurko.jvm_bloggers.utils.DateTimeUtilities.lastPublicationDate;
+
+/**
+ * @autor mszarlinski
+ */
+@Component
+public class BlogPostItemPopulator {
+    private static final DateTimeFormatter DATE_FORMATTER
+            = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+
+    private static final String ACTION_PANEL_WICKET_ID = "actionPanel";
+
+    @Autowired
+    private NowProvider nowProvider;
+
+    public void populateItem(final Item<BlogPost> item, final Form<Void> moderationForm,
+                             final CustomFeedbackPanel feedbackPanel) {
+        final BlogPost post = item.getModelObject();
+        item.add(new Label("title", post.getTitle()));
+        item.add(new Label("author", post.getBlog().getAuthor()));
+        item.add(new ExternalLink("link", post.getUrl(), abbreviate(post.getUrl(), 90)));
+        item.add(new Label("date", post.getPublishedDate().format(DATE_FORMATTER)));
+        item.add(new Label("approved", post.getApprovalState()));
+        addEvenOddRowStyling(item);
+
+        applyPostSpecificRowStyles(item, moderationForm,
+                post.isGoingInNewsletter(lastPublicationDate(nowProvider.now())), feedbackPanel);
+    }
+
+    @VisibleForTesting
+    void applyPostSpecificRowStyles(final Item<BlogPost> item, final Form<Void> moderationForm,
+                                    final boolean postIsGoingInNewsletter,
+                                    final CustomFeedbackPanel feedbackPanel) {
+        if (postIsGoingInNewsletter) {
+            item.add(new ModerationActionPanel(ACTION_PANEL_WICKET_ID, moderationForm,
+                    feedbackPanel, item.getModel()));
+            addHighlightedRowStyling(item);
+        } else {
+            // hide moderation buttons
+            item.add(new EmptyPanel(ACTION_PANEL_WICKET_ID));
+        }
+    }
+
+    private void addEvenOddRowStyling(final Item<BlogPost> item) {
+        item.add(AttributeModifier.append("class", new AbstractReadOnlyModel<String>() {
```

Comment:
This line causes:
`org.springframework.core.serializer.support.SerializationFailedException: Failed to serialize object using DefaultSerializer; nested exception is java.io.NotSerializableException: pl.tomaszdziurko.jvm_bloggers.view.admin.moderation.BlogPostItemPopulator`